### PR TITLE
fix: naming and storybook type issue workaround

### PR DESCRIPTION
**Pull request by JoCa96**
#222 

- Implement decisison to not prefix boolean props.
- Propose workaround for storybook issue

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxRadioButton/OnyxRadioButton.stories.ts [hunk 1]
```diff
@@ -10,13 +10,7 @@ const meta: Meta<typeof OnyxRadioButton> = {
...
     component: OnyxRadioButton,
     events: ["update:modelValue"],
-    argTypes: {
-      isReadonly: { type: "boolean" },
-      isLoading: { type: "boolean" },
-      isDisabled: { type: "boolean" },
-      selected: { type: "boolean" },
-      errorMessage: { type: "string" },
-    },
+    argTypes: {},
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxRadioButton/OnyxRadioButton.stories.ts:13]: Can this prop be removed?

**Review by larsrickert — APPROVED** [review]

**Review by larsrickert — APPROVED** [review]

File: packages/sit-onyx/src/components/OnyxRadioButton/types.ts [hunk 2]
```diff
@@ -10,9 +10,15 @@ export type SelectionOption<T> = {
...
+  disabled?: boolean;
+  readonly?: boolean;
+  loading?: boolean;
 };
 
 export type SelectionProps<T> = SelectionOption<T> & { selected?: boolean };
+
+export type RadioButtonProps<TValue> = SelectionProps<TValue> & {
+  name: string;
+  isRequired?: boolean;
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxRadioButton/types.ts:22]:
```suggestion
  required?: boolean;
```

File: packages/sit-onyx/src/types/utils.ts [hunk 3]
```diff
@@ -2,3 +2,6 @@
  * Recursive / deep implementation of TypeScript's built-in `Partial<T>` type.
  */
 export type DeepPartial<T> = T extends object ? { [P in keyof T]?: DeepPartial<T[P]> } : T;
+
+export type Equals<A extends never> = A;
```
Comment (BoppLi) [inline packages/sit-onyx/src/types/utils.ts:6]: do you also want to leave the comment to remove it wiht the workaround or can this also be useful for other use cases? :)

**Review by BoppLi — DISMISSED** [review]

**Comment by BoppLi** [comment]
> #222

is it possible the ticket is not properly linked?
image
this is empty? intentionally?

**Review by BoppLi — DISMISSED** [review]

**Review by larsrickert — APPROVED** [review]

**Pull request merged**